lib/src/components/camera/service.dart

```diff
@@ -0,0 +1,61 @@
+import 'package:grpc/grpc.dart';
+import 'package:viam_sdk/src/gen/google/api/httpbody.pb.dart';
+
+import '../../gen/common/v1/common.pb.dart';
+import '../../gen/component/camera/v1/camera.pbgrpc.dart';
+import '../../media/image.dart';
+import '../../resource/manager.dart';
+import '../../utils.dart';
+import 'camera.dart';
+
+class CameraService extends CameraServiceBase {
+  final ResourceManager _manager;
+
+  CameraService(this._manager);
+
+  Camera _fromManager(String name) {
+    try {
+      return _manager.getResource(Camera.getResourceName(name));
+    } catch (e) {
+      throw (GrpcError.notFound(e.toString()));
+    }
+  }
+
+  @override
+  Future<DoCommandResponse> doCommand(ServiceCall call, DoCommandRequest request) async {
+    final camera = _fromManager(request.name);
+    final result = await camera.doCommand(request.command.toMap());
+    return DoCommandResponse(result: result.toStruct());
+  }
+
+  @override
+  Future<GetImageResponse> getImage(ServiceCall call, GetImageRequest request) async {
+    final camera = _fromManager(request.name);
+    final image = await camera.image(mimeType: MimeType.fromString(request.mimeType));
+    return GetImageResponse(mimeType: image.mimeType.toString(), image: image.raw);
+  }
+
+  @override
+  Future<GetPointCloudResponse> getPointCloud(ServiceCall call, GetPointCloudRequest request) async {
+    final camera = _fromManager(request.name);
+    final image = await camera.pointCloud();
+    return GetPointCloudResponse(mimeType: image.mimeType.toString(), pointCloud: image.raw);
+  }
+
+  @override
+  Future<GetPropertiesResponse> getProperties(ServiceCall call, GetPropertiesRequest request) async {
+    final camera = _fromManager(request.name);
+    final properties = await camera.properties();
+    return GetPropertiesResponse(
+        supportsPcd: properties.supportsPcd,
+        intrinsicParameters: properties.intrinsicParameters,
+        distortionParameters: properties.distortionParameters);
+  }
+
+  @override
+  Future<HttpBody> renderFrame(ServiceCall call, RenderFrameRequest request) async {
+    final camera = _fromManager(request.name);
+    final image = await camera.image(mimeType: MimeType.fromString(request.mimeType));
+    return HttpBody(data: image.raw, contentType: image.mimeType.toString());
+  }
```

Comment:
Q: are planning to convert our custom rbga images here or in the client (or just punting for now)

Reply:
CC @njooma since you might have more context here

Reply:
I don't understand what you're asking.

I tried to copy what this method does in the python and cpp sdk for this method, but not doing any `converting` that i know of.

Reply:
so we have a custom image mime type called `image/vnd.viam.rgba`, that requires custom logic to decode (e.g. [here's the logic that does that in python](https://github.com/viamrobotics/viam-python-sdk/blob/7ae6a3cbc4f80295c61a4104a751a35d0d94836a/src/viam/media/viam_rgba_plugin.py))

however i think i know the answer to my own question - we need to decode it to display it properly so this logic will be applied client-side

Reply:
@maximpertsov I already took care of ViamRGBA type [here](https://github.com/viamrobotics/viam-flutter-sdk/blob/1f8acb5a19984e6100b2b698ceaa051c45786aa9/lib/src/media/image.dart#L107). Everything is taken care of under the hood :)